interface/ceed-operator.c

```diff
@@ -2106,6 +2110,35 @@ int CeedOperatorApplyAdd(CeedOperator op, CeedVector in, CeedVector out, CeedReq
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Destroy temporary assembly data associated with a `CeedOperator`
+
+  @param[in,out] op `CeedOperator` whose assembly data to destroy
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedOperatorAssemblyDataStrip(CeedOperator op) {
```

Comment:
Can we call this in `CeedOperatorDestroy` to make sure it does what we think it does?

Reply:
Done in [f883f0a](https://github.com/CEED/libCEED/pull/1619/commits/f883f0a5423dccfdefcc45bb5c497717e44013fc). One nuance here is that in `CeedOperatorAssemblyDataStrip` I am destroying assembly for all sub-operators of a composite operator which is useful for external users. This just means we need to call it before destroying the sub-operators.